packages/react-component-library/src/components/DatePickerE/DatePickerE.tsx

```diff
@@ -153,20 +161,20 @@ export const DatePickerE: React.FC<DatePickerEProps> = ({
       >
         <StyledOuterWrapper
           data-testid="datepicker-outer-wrapper"
-          $hasFocus={open}
+          $hasFocus={hasFocus && !hasError}
```

Comment:
The `&& !hasError` is temporary here to preserve the existing behaviour. It will be reviewed as part of #2863.